File: leetcode/math/728-Self-Dividing-Numbers.py
class Solution:
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        return [number for number in range(left, right+1) if '0' not in str(number) and all((number % int(char) == 0 for char in str(number)))]

File: leetcode/math/728-Self-Dividing-Numbers.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> ans;
        for(int i = left; i <= right; i++) {
            int temp = i;
            vector<int> res;
            while(temp > 0) {
                int digit = temp % 10;
                temp /= 10;
                res.push_back(digit);
            }
            bool flag = true;
            for(int j = 0; j < res.size(); j++) {
                if(res[j] == 0 || i % res[j] != 0) {
                    flag = false;
                    break;
                }
            }
            if(flag) ans.push_back(i);
        }
        return ans;
    }
};